lua/wikis/valorant/MatchGroup/Input/Custom.lua

```diff
@@ -137,6 +157,152 @@ function MatchFunctions.getPatch(match, games)
+			clutches = 0,
+		}
+	)
+end
+
+---@param maps table[]
+---@param player table
+---@param teamIdx number
+---@return table
+function MatchFunctions.calculateOverallStatsForPlayer(maps, player, teamIdx)
+	local playerId = player.player
+	if not playerId then return {} end
+
+	local overallStats = {
+		acs = {},
+		kast = {},
+		adr = {},
+		kills = 0,
+		deaths = 0,
+		assists = 0,
+		firstKills = 0,
+		firstDeaths = 0,
+		totalRoundsPlayed = 0,
+		totalKastRounds = 0,
+		damageDealt = 0,
+	}
+	local agents = {}
+
+	Array.forEach(maps, function(map)
+		if map.status == MatchGroupInputUtil.MATCH_STATUS.NOT_PLAYED then
+			return
+		end
+
+		local mapOpponent = map.opponents[teamIdx]
+		if not mapOpponent or not mapOpponent.players then
+			return
+		end
+
+		local mapPlayerIndex
+
+		local mapPlayer = Array.find(mapOpponent.players, function(playerData, playerIdx)
+			mapPlayerIndex = playerIdx
+			return playerData.player == playerId
+		end)
+
+		if not mapPlayer then
+			return
+		end
+
+		if mapPlayer.agent then
+			table.insert(agents, mapPlayer.agent)
+		end
+		if mapPlayer.acs then table.insert(overallStats.acs, mapPlayer.acs) end
+		if mapPlayer.kast then table.insert(overallStats.kast, mapPlayer.kast) end
+		if mapPlayer.adr then table.insert(overallStats.adr, mapPlayer.adr) end
+		overallStats.kills = overallStats.kills + (mapPlayer.kills or 0)
+		overallStats.deaths = overallStats.deaths + (mapPlayer.deaths or 0)
+		overallStats.assists = overallStats.assists + (mapPlayer.assists or 0)
+		overallStats.totalRoundsPlayed = overallStats.totalRoundsPlayed + (mapPlayer.totalRounds or 0)
+		overallStats.totalKastRounds = overallStats.totalKastRounds + (mapPlayer.kastRounds or 0)
+		overallStats.damageDealt = overallStats.damageDealt + (mapPlayer.damageDealt or 0)
+
+		local extraDataPlayer = Array.find(map.extradata.teams[teamIdx].players, function(playerData)
+			return playerData.player == playerId
+		end)
+		if not extraDataPlayer then
+			extraDataPlayer = map.extradata.teams[teamIdx].players[mapPlayerIndex]
+		end
+
+		overallStats.firstKills = overallStats.firstKills + (extraDataPlayer.firstKills or 0)
+		overallStats.firstDeaths = overallStats.firstDeaths + (extraDataPlayer.firstDeaths or 0)
+	end)
+
+	local function average(statTable)
+		if #statTable == 0 then return nil end
+		local sum = Array.reduce(statTable, Operator.add)
+		return sum / #statTable
+	end
+
+	local function calculatePercentage(value, total)
+		if total == 0 then
+			return 0
+		end
+		return value / total * 100
+	end
+
+	return {
+		teamIndex = teamIdx,
+		player = player.player,
+		displayName = player.displayName,
+		agent = agents,
+		acs = average(overallStats.acs),
+		kills = overallStats.kills,
+		deaths = overallStats.deaths,
+		assists = overallStats.assists,
+		kast = calculatePercentage(overallStats.totalKastRounds, overallStats.totalRoundsPlayed) or nil,
+		adr = overallStats.damageDealt / overallStats.totalRoundsPlayed or nil,
```

Comment:
Not sure when the or nil would be applicable here. 

Also verify that totalRoundsPlayer is better than 0